builtin/providers/openstack/resource_openstack_networking_secgroup_rule_v2.go

```diff
@@ -83,6 +88,16 @@ func resourceNetworkingSecGroupRuleV2Create(d *schema.ResourceData, meta interfa
 		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
 	}
 
+	portRangeMin := d.Get("port_min_range").(string)
+	portRangeMax := d.Get("port_min_range").(string)
+	protocol := d.Get("protocol").(string)
+
+	if protocol == "" {
+		if portRangeMin != "" || portRangeMax != "" {
```

Comment:
Github is being a little funny, so I'll repeat the last comments... It looks like the rough sketch I made had some errors. After testing actual code, the following should work:

```
portRangeMin := d.Get("port_range_min").(int)
portRangeMax := d.Get("port_range_max").(int)
protocol := d.Get("protocol").(string)

if protocol == "" {
	if portRangeMin != 0 || portRangeMax != 0 {
```